pkg/core/interop_neo_test.go

```diff
@@ -0,0 +1,435 @@
+	script := v.Estack().Pop().Value()
+	require.Equal(t, contractState.Script, script)
+}
+
+func TestContractIsPayable(t *testing.T) {
+	v, contractState, context := createVMAndContractState(t)
+	v.Estack().PushVal(vm.NewInteropItem(contractState))
+
+	err := context.contractIsPayable(v)
+	require.NoError(t, err)
+	isPayable := v.Estack().Pop().Value()
+	require.Equal(t, contractState.IsPayable(), isPayable)
+}
+
+func TestAssetGetAdmin(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAdmin(v)
+	require.NoError(t, err)
+	admin := v.Estack().Pop().Value()
+	require.Equal(t, assetState.Admin.Bytes(), admin)
+}
+
+func TestAssetGetAmount(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAmount(v)
+	require.NoError(t, err)
+	amount := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.Amount)), amount)
+}
+
+func TestAssetGetAssetID(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAssetID(v)
+	require.NoError(t, err)
+	assetID := v.Estack().Pop().Value()
+	require.Equal(t, assetState.ID.Bytes(), assetID)
+}
+
+func TestAssetGetAssetType(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAssetType(v)
+	require.NoError(t, err)
+	assetType := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.AssetType)), assetType)
+}
+
+func TestAssetGetAvailable(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAvailable(v)
+	require.NoError(t, err)
+	available := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.Available)), available)
+}
+
+func TestAssetGetIssuer(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetIssuer(v)
+	require.NoError(t, err)
+	issuer := v.Estack().Pop().Value()
+	require.Equal(t, assetState.Issuer.Bytes(), issuer)
+}
+
+func TestAssetGetOwner(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetOwner(v)
+	require.NoError(t, err)
+	owner := v.Estack().Pop().Value()
+	require.Equal(t, assetState.Owner.Bytes(), owner)
+}
+
+func TestAssetGetPrecision(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetPrecision(v)
+	require.NoError(t, err)
+	precision := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.Precision)), precision)
+}
+
+func TestRuntimeSerializeDeserialize(t *testing.T) {
```

Comment:
Isn't it a `vm` test?

Reply:
we have interop serialize and deserialize functions:
```
// RuntimeSerialize handles syscalls System.Runtime.Serialize and Neo.Runtime.Serialize.
func RuntimeSerialize(vm *VM) error {
...
```
and
```
// RuntimeDeserialize handles syscalls System.Runtime.Deserialize and Neo.Runtime.Deserialize.
func RuntimeDeserialize(vm *VM) error {
```
that's what we test right here

Reply:
Sure, but all the real code is hidden in the `vm` package, so this test should probably be moved there, because `core` only has has thin wrappers around that (you can test that, of course, but you need to call them and not `vm.*` then).

Reply:
I see.
Can't just move them since for making random data we need access from that package to other packages which leads to circular deps. So just removed them

Reply:
Several options here:
 * use `*_test` package
 * move `random_util_test.go` into some `pkg/internal/testrandom` and use that where appropriate

Reply:
Although I guess that `vm` part is already tested well, so maybe it's just not worth it. But you can add these functions into the "Missing tests" lists then.

Reply:
added to missing tests section